### PR TITLE
Firebase trigger a

### PR DESCRIPTION
**Pull request by Arun333333**

### CODE REVIEW

File: .github/actions/firebase-trigger/action.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: 'Github trigger'
```
Comment (Robert483) [inline .github/actions/firebase-trigger/action.yml:1]:
```suggestion
name: 'Firebase Trigger'
```
Rename this one

File: .github/actions/firebase-trigger/action.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+description: 'Gets the data from other action and changes the firestore'
+inputs:
+  credentials:
+    description: 'Firebase credentials'
+    required: true
+    
+  collection:  # id of input
+    description: 'name of the collection the data is being added'
+    required: true
+    default: 'filters'
```
Comment (Robert483) [inline .github/actions/firebase-trigger/action.yml:11]:
```suggestion
    default: 'database'
```
we have new structure

File: .github/actions/firebase-trigger/action.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    
+  doc: # name of document
+    description: 'name of the document'
+    required: true
+    default: 'database'
```
Comment (Robert483) [inline .github/actions/firebase-trigger/action.yml:16]:
```suggestion
    default: 'filters'
```
same here. We have new structure

File: .github/actions/firebase-trigger/action.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  
+  value: #json object
+    description: 'json object'
```
Comment (Robert483) [inline .github/actions/firebase-trigger/action.yml:19]:
```suggestion
    description: 'the value to be added to firestore'
```
the old description is not clear

File: .github/actions/firebase-trigger/index.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+const core = require('@actions/core');
+const github = require('@actions/github');
+const admin = require('firebase-admin');
+const firebase = require('firebase');
```
Comment (Robert483) [inline .github/actions/firebase-trigger/index.js:4]: I just found out that we only need firebase-admin. We can remove this line, and perform 'npm uninstall firebase'

File: .github/actions/firebase-trigger/index.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		credential: admin.credential.cert(JSON.parse(credentials)),
+		databaseURL: core.getInput('databaseUrl'),
+	});
+}
+// This function gets the output of the other action to get the github files URL.
+function getValue() {
+	core.info('Getting the value');
+	const value = core.getInput('value');
+	return value;
+}
```
Comment (Robert483) [inline .github/actions/firebase-trigger/index.js:19]: This only part that matters is core.getInput('value'), so we can delete this function and use it inside processJavascriptAction()

File: .github/actions/firebase-trigger/index.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+// main function of the script calls all the functions.
+function processJavascriptAction() {
+	initFirebase();
+	const collection = core.getInput('collection');
+	const value = getValue();
+	var jsonValue = JSON.parse(value);
+	updateFirestoreDatabase(collection, jsonValue);
+}
+processJavascriptAction();
```
Comment (Robert483) [inline .github/actions/firebase-trigger/index.js:41]:
```suggestion
function run() {
	initFirebase();
	const collection = core.getInput('collection');
	const value = getValue();
	var jsonValue = JSON.parse(value);
	updateFirestoreDatabase(collection, jsonValue);
}

run();
```
We should change the name of the function to make it consistent with my action

**Review by bryan7zhang — APPROVED** [review]

**Pull request merged**